tests/test_opnieuw.py

```diff
@@ -65,9 +66,44 @@ def test_retry_times(self):
         except TypeError as e:
             end = time.time()
             t_diff = end - start
-            self.assertGreater(t_diff, 1)
+            self.assertGreater(t_diff, 0.5)
             self.assertEqual(self.counter, 3)
 
+    def test_retry_immediately_global(self) -> None:
+        start = time.time()
+        self.counter = 0
+        with retry_immediately():
+            try:
+                self.foo()
+            except TypeError as e:
+                end = time.time()
+                t_diff = end - start
```

Comment:
If we include the unit in the name, for example by naming this `seconds_elapsed` instead of `t_diff`, then it is clear that the 0.5 on the following line is 0.5 seconds, and not 0.5 milliseconds, or 0.5 Martian days.

I happen to remember that `time.time()` returns float seconds now, but for a reader who does not know this, milliseconds would be a reasonable guess. For a while I was doing Scala, Haskell, and Python regularly, and in JVM world, `Thread.sleep` takes milliseconds, in Haskell `threadDelay` takes microseconds, and in Python `time.sleep` takes seconds, so I always had to consult the docs to avoid mixing them up.